fdbclient/NativeAPI.actor.cpp

```diff
@@ -4303,7 +4310,9 @@ Future<RangeResultFamily> getRange(Reference<TransactionState> trState,
 			req.arena.dependsOn(mapper.arena());
 			setMatchIndex<GetKeyValuesFamilyRequest>(req, matchIndex);
 			req.tenantInfo = useTenant ? trState->getTenantInfo() : TenantInfo();
-			req.isFetchKeys = (trState->taskID == TaskPriority::FetchKeys);
+			req.options = trState->readOptions;
+			// Don't cache for fetch
+			req.options.cacheResult = (req.options.type != ReadType::FETCH);
```

Comment:
I think we shouldn't set this here, and rely on the users of Native Client to set it.  While for practical purposes a FETCH read should have cacheResult set false, let's set that at the source of the fetch reads which is FetchKeys in StorageServer.